### ISSUE TITLE
Clone with New Name can misbehave on folders on storageV2

### ISSUE THREAD

**Issue by narfanar** [microsoft/AzureStorageExplorer#3243]
**Storage Explorer Version**: 1.31.1
**Build Number**: 20200515.1
**Platform/OS**: Windows 10 v1909
**Architecture**: x64
**Regression From**: Unknown

## Bug Description ##
When "cloning" a folder with a new name on a storageV2 account container, a weird blob is created with the new name and the size of the last file that was added to the folder being cloned.

There's no new folder created.

And obviously no files have been copied as there's no folder to put them.

## Steps to Reproduce ##
1. Go to a container on storageV2
2. Upload some files in a nested folder. Note the size of the last file uploaded.
3. Go back up a level and select the folder
4. "Clone with New Name"
5. See a binary blob appear with the new name and the size of the last file you uploaded to the folder.
6. No folder is created.

## Expected Experience ##
Clone with New Name on folders clones the entire structure.

## Actual Experience ##
Clone with New Name on folders is misbehaving. Current behavior doesn't seem to be of much use, and is likely not intentional.

## Additional Context ##
N/A

**Comment by JasonYeMSFT** [comment]
@narfanar I have tried the same steps from 1.13.1 to 1.14.2 but cannot reproduce this issue. Could you tell us the name of the folder you are trying to clone?

**Comment by narfanar** [comment]
Some more details:

- The source folder was not at root level
- Source folder had only binary/non-text files
- Source folder and files were created through a blobfuse mount of the container
- Mount was on Ubuntu 18.04 be the creation steps were:
  - mkdir the nested path (**important**)
  - cp files to leaf folder

Just a hunch, but the mkdir seems like a good starting point.

I'll try to repro it as well.

**Comment by JasonYeMSFT** [comment]
I have tried the blobfuse . My repro steps:
1. Mount a container via blobfuse
2. mkdir "newFolder"

What I see then is that a binary blob with name "newFolder" and metadata "hdi_isFolder=true" is created. This is a stub blob for an empty folder.

3. cp "some_file" "path_to_mounted_directory"

What I see is that there will be a folder with name "newFolder" next to the "newFolder" stub blob.
If I "clone with new name" on the folder named "newFolder" to "newFolder2", a folder named "newFolder2" will be created next to the folder "newFolder", containing contents under "newFolder". This behavior is expected.

If I "clone with new name" on the stub blob named "newFolder" to "newFolder2", a folder named "newFolder2" will be created next to the folder "newFolder", containing a folder named "newFolder". That nested "newFolder" contains the content of the original "newFolder". This is an unexpected operation that we don't support for now.

To summarize, our current implementation doesn't fully support such special folders with stub blobs. Please avoid manipulating such folders in Storage Explorer for now. As a workaround, you can use the mv command within your mounted container to achieve the renaming.

**Comment by JasonYeMSFT** [comment]
Closed due to lack of response. If you are still having issue with it, please open a new issue and share your thoughts/findings if what you see is different from my investigation result. Thanks.
